internal/grpc/services/gateway/storageprovider.go

```diff
@@ -1089,42 +1113,42 @@ func (s *svc) Move(ctx context.Context, req *provider.MoveRequest) (*provider.Mo
 }
 
 func (s *svc) move(ctx context.Context, req *provider.MoveRequest) (*provider.MoveResponse, error) {
-	srcList, err := s.findProviders(ctx, req.Source)
+	srcProviders, err := s.findProviders(ctx, req.Source)
 	if err != nil {
 		return &provider.MoveResponse{
 			Status: status.NewStatusFromErrType(ctx, "move src="+req.Source.String(), err),
 		}, nil
 	}
 
-	dstList, err := s.findProviders(ctx, req.Destination)
+	dstProviders, err := s.findProviders(ctx, req.Destination)
 	if err != nil {
 		return &provider.MoveResponse{
 			Status: status.NewStatusFromErrType(ctx, "move dst="+req.Destination.String(), err),
 		}, nil
 	}
 
 	// if providers are not the same we do not implement cross storage move yet.
-	if len(srcList) != 1 || len(dstList) != 1 {
+	if len(srcProviders) != 1 || len(dstProviders) != 1 {
 		res := &provider.MoveResponse{
 			Status: status.NewUnimplemented(ctx, nil, "gateway: cross storage copy not yet implemented"),
 		}
 		return res, nil
 	}
 
-	srcP, dstP := srcList[0], dstList[0]
+	srcProvider, dstProvider := srcProviders[0], dstProviders[0]
 
 	// if providers are not the same we do not implement cross storage copy yet.
-	if srcP.Address != dstP.Address {
+	if srcProvider.Address != dstProvider.Address {
```

Comment:
@butonic, If I understand correctly than this shouldn't be a cross storage copy/move. I didn't remove this check here.
Or is a move between two shares always considered a cross storage copy/move

Reply:
ok, now I get it. This fixes the case for moving a file from one share into another share from the same storage, eg.
`/home/Shares/foo/readme.txt` -> `/home/Shares/bar/readme.md`, where `foo` and `bar` are located on the same storage provider.

@labkode then this is good to merge IMO.

Reply:
Yes, exactly. :+1: